eternalcore-core/src/main/java/com/eternalcode/core/feature/ignore/IgnoreServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package com.eternalcode.core.feature.ignore;
+
+import com.eternalcode.core.event.EventCaller;
+import com.eternalcode.core.feature.ignore.event.IgnoreAllEvent;
+import com.eternalcode.core.feature.ignore.event.IgnoreEvent;
+import com.eternalcode.core.feature.ignore.event.UnIgnoreAllEvent;
+import com.eternalcode.core.feature.ignore.event.UnIgnoreEvent;
+import org.bukkit.Server;
+import org.bukkit.entity.Player;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+
+public class IgnoreServiceImpl implements IgnoreService {
+
+    private final IgnoreRepository ignoreRepository;
+    private final Server server;
+    private final EventCaller caller;
+
+    public IgnoreServiceImpl(IgnoreRepository ignoreRepository, Server server, EventCaller caller) {
+        this.ignoreRepository = ignoreRepository;
+        this.server = server;
+        this.caller = caller;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> isIgnored(UUID by, UUID target) {
+        return null;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> ignore(UUID by, UUID target) {
+        Player senderPlayer = this.server.getPlayer(by);
+        Player targetPlayer = this.server.getPlayer(target);
+        IgnoreEvent event = this.caller.callEvent(new IgnoreEvent(senderPlayer, targetPlayer));
+
+        if (event.isCancelled()) {
+            return CompletableFuture.completedFuture(false);
+        }
+
+        this.ignoreRepository.ignore(by, target);
+        return CompletableFuture.completedFuture(true);
```

Comment:
połącz to jakoś tak aby completable miało sens i uzupełniło się o true po wykonaniu operacji na bazie danych

`return this.ignoreRepository.ignore(by, target).thenApply(() -> true);`